src/ethereum/EthereumEngine.ts

```diff
@@ -144,9 +144,10 @@ export class EthereumEngine extends CurrencyEngine<
 
     this.utils = {
       signMessage: (message: string, privateKeys: EthereumPrivateKeys) => {
-        if (!isHex(message)) throw new Error('ErrorInvalidMessage')
         const privKey = Buffer.from(privateKeys.privateKey, 'hex')
-        const messageBuffer = hexToBuf(message)
+        const messageBuffer = isHex(message)
+          ? hexToBuf(message)
+          : Buffer.from(message)
```

Comment:
I'm not sure it's a good idea to use Buffer.from without specifying the encoding. Instead maybe explicitly test for different types of strings and use base16 and base64 to create the uint8array and to support arbitrary text by using  Buffer.from(arg, 'ascii')